staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java

```diff
@@ -234,4 +244,91 @@ private void completeInProgressBatch(final TenantId tenantId, final Path inProgr
+    {
+        if (skipBatchFileCleanup) {
+            return;
+        }
+        try {
+            final List<Path> batchesToClean = Files.list(Paths.get(basePath))
+                .map(p -> getTenantInprogressDirectorySafely(p.getFileName().toString()))
+                .flatMap(p -> getAllBatchFilesForAllDirectories(p.get()))
+                .filter(p -> BatchNameProvider.validateFileName(p.getFileName().toString()))
+                .filter(p -> shouldDelete(p.getFileName().toString()))
+                .filter(p -> checkAllSubfilesSafely(p))
+                .collect(Collectors.toList());
+            for (final Path filepath : batchesToClean) {
+                try {
+                    FileUtils.deleteDirectory(filepath.toFile());
+                } catch (final IOException | IllegalArgumentException ex) {
+                    LOGGER.error("Unable to delete directory {}", filepath);
+                    LOGGER.debug("An error occured while attempting to delete folder {}", filepath, ex);
+                }
+            }
+        } catch (final IOException ex) {
+            LOGGER.error("An exception occured trying to read the files in the base directory.", ex);
+        }
+    }
+
+    private boolean checkAllSubfilesSafely(final Path path)
+    {
+        try {
+            return Files.list(path)
+                .filter(p -> BatchNameProvider.validateFileName(p.getFileName().toString()))
+                .filter(p -> !shouldDelete(p.getFileName().toString()))
+                .collect(Collectors.toList())
+                .isEmpty();
+        } catch (final IOException ex) {
+            LOGGER.error("Unable to open directory {}", path, ex);
+            return false;
+        }
+    }
+
+    /**
+     * Obtain a path to the tenants in_progress folder, if the string provided is not a valid tenant id this method will return 
+     * an empty Optional
+     * @param tenantIdFolderName The name of the tenant
+     * @return An Optional of paths to the tenants in_progress folder
+     */
+    private Optional<Path> getTenantInprogressDirectorySafely(final String tenantIdFolderName)
+    {
+        try{
+            final TenantId tenantId = new TenantId(tenantIdFolderName);
+            return Optional.ofNullable(batchPathProvider.getTenantInprogressDirectory(tenantId));
+        } catch (final InvalidTenantIdException ex){
+            LOGGER.debug("Ignoring folder {} as it does not represent a valid tenantId.", tenantIdFolderName);
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Returns a list of the path to all sub files in all folders provide
+     * @param tenantFolders List of Path objects that represent all current tenant in_progress folders
+     * @return return a Stream of paths to all subfiles under all folders provided
+     */
+    private Stream<Path> getAllBatchFilesForAllDirectories(final Path tenantFolder)
+    {
+        try {
+            return Files.list(tenantFolder);
```

Comment:
* <p> The returned stream encapsulates one or more {@link DirectoryStream}s.
     * If timely disposal of file system resources is required, the
     * {@code try}-with-resources construct should be used to ensure that the

Reply:
If I do that wont the steam be closed once this method exits? I want the stream returned to the caller to operate on.